core/exceptions.py

```diff
@@ -1,4 +1,14 @@
+from core.messaging import Message
+
+
 class ValidationError(RuntimeError):
+    error_messages: list[Message]
+
+    def __init__(self, error_messages: list[Message]):
+        self.error_messages = error_messages
+
+
+class OldValidationError(RuntimeError):
```

Comment:
should we add some context here to explain what we mean by "old.." at least a TODO with a Jira ref to deal with it